### PR TITLE
PORTALS-3151, PORTALS-3152

### PR DESCRIPTION
**Pull request by nickgros**
- Fix URL when resetting password from synapse.org
- Only allow sending one reset password request per email/username

### CODE REVIEW

File: apps/SageAccountWeb/src/components/ResetPassword.tsx [hunk 1]
```diff
@@ -38,6 +40,7 @@ export const ResetPassword = (props: ResetPasswordProps) => {
     clickEvent.preventDefault()
     try {
       await SynapseClient.resetPassword(userName)
+      setHasClickedResetPassword(true)
```
Comment (nickgros) [inline apps/SageAccountWeb/src/components/ResetPassword.tsx:43]: When the REST call succeeds, disable the button

File: apps/SageAccountWeb/src/components/ResetPassword.tsx [hunk 2]
```diff
@@ -84,7 +87,10 @@ export const ResetPassword = (props: ResetPasswordProps) => {
                 label={'Email address or username'}
                 id="username"
                 name="username"
-                onChange={e => setUserName(e.target.value)}
+                onChange={e => {
+                  setHasClickedResetPassword(false)
```
Comment (nickgros) [inline apps/SageAccountWeb/src/components/ResetPassword.tsx:91]: if the user changes the username/email in the text box, re-enable the button

File: packages/synapse-react-client/src/components/Authentication/LoginForm.tsx [hunk 3]
```diff
@@ -66,8 +55,9 @@ export default function LoginForm(props: Props) {
     hideForgotPasswordButton,
     twoFactorAuthResetIsSuccess,
     twoFactorAuthResetIsPending,
-    twoFactorAuthResetUri = appendTwoFAResetTokenQueryParam(
+    twoFactorAuthResetUri = appendFinalQueryParamKey(
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/Authentication/LoginForm.tsx:58]: refactor to reusable function

File: packages/synapse-react-client/src/utils/appendFinalQueryParamKey.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * such as initiating a reset password request or reset 2FA request.
+ * @param url
+ * @param queryParam
+ */
+export default function appendFinalQueryParamKey(
+  url: URL,
+  queryParam: string,
+): string {
+  // TODO: URLSearchParams.size returns undefined in our test environment
+  if (Array.from(url.searchParams).length === 0) {
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/appendFinalQueryParamKey.ts:14]: Upgrading JSDOM alone did not fix this

File: packages/synapse-react-client/src/synapse-client/SynapseClient.ts [hunk 5]
```diff
@@ -4204,7 +4205,10 @@ export const changePassword = (
 
 // http://rest-docs.synapse.org/rest/POST/user/password/reset.html
 export const resetPassword = (email: string) => {
-  const endpoint = window.location.href + '?passwordResetToken='
+  const endpoint = appendFinalQueryParamKey(
+    new URL(window.location.href),
+    'passwordResetToken',
+  )
```
Comment (nickgros) [inline packages/synapse-react-client/src/synapse-client/SynapseClient.ts:4211]: Fixes PORTALS-3151

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**